iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -659,19 +660,40 @@ popo::InterfacePortData* PortManager::acquireInterfacePortData(capro::Interfaces
     }
 }
 
+void PortManager::publishServiceRegistry() const noexcept
+{
+    if (m_serviceRegistryPublisherPortData.has_value())
+    {
+        PublisherPortUserType publisher(m_serviceRegistryPublisherPortData.value());
+        publisher
+            .tryAllocateChunk(sizeof(ServiceRegistry),
+                              alignof(ServiceRegistry),
+                              CHUNK_NO_USER_HEADER_SIZE,
+                              CHUNK_NO_USER_HEADER_ALIGNMENT)
+            .and_then([&](auto& chunk) {
```

Comment:
Here we should also print at least a warning if we cannot allocate another chunk.